src/openrct2/actions/PlaceParkEntranceAction.hpp

```diff
@@ -79,13 +79,12 @@ DEFINE_GAME_ACTION(PlaceParkEntranceAction, GAME_COMMAND_PLACE_PARK_ENTRANCE, Ga
 
         int8_t zLow = _loc.z / 8;
         int8_t zHigh = zLow + 12;
-        LocationXY16 entranceLoc = { (int16_t)_loc.x, (int16_t)_loc.y };
+        CoordsXYZ entranceLoc = _loc;
         for (uint8_t index = 0; index < 3; index++)
         {
             if (index == 1)
             {
-                entranceLoc.x += CoordsDirectionDelta[(_loc.direction - 1) & 0x3].x;
-                entranceLoc.y += CoordsDirectionDelta[(_loc.direction - 1) & 0x3].y;
+                entranceLoc += CoordsDirectionDelta[(_loc.direction - 1) & 0x3];
             }
             else if (index == 2)
             {
```

Comment:
You can change L91 as well

Reply:
I can't there is a multiplication first, only if I implement `operator*`